File: codewars/cpp/get_the_mean_of_an_array.cpp
// https://www.codewars.com/kata/563e320cee5dddcf77000158/train/cpp


#include <iostream>
#include <vector>
#include <numeric>

int get_average(std::vector<int> marks) {
    return std::accumulate(marks.begin(), marks.end(), 0) / marks.size();
}

int main() {
    int lim, val;
    std::vector<int> marks;

    std::cin >> lim;
    for(int i = 0; i < lim; ++i) {
        std::cin >> val;
        marks.push_back(val);
    }

    std::cout << get_average(marks) << "\n";

    return 0;
}

File: codewars/elixir/get_the_mean_of_an_array.exs
# https://www.codewars.com/kata/563e320cee5dddcf77000158/train/elixir

defmodule Calculator do
  def get_average(marks) do
    Kernel.div(Enum.sum(marks), Kernel.length(marks))
  end
end


IO.puts(Calculator.get_average([2,2,2,2]))
IO.puts(Calculator.get_average([1,2,3,4,5,]))
IO.puts(Calculator.get_average([1,1,1,1,1,1,1,2]))


